narwhals/_pandas_like/series.py

```diff
@@ -643,13 +644,30 @@ def to_dummies(
         plx = self.__native_namespace__()
         series = self._native_series
         name = str(self._name) if self._name else ""
+
+        null_col_pl = f"{name}{separator}null"
+
+        has_nulls = series.isna().any()
+        result = plx.get_dummies(
+            series,
+            prefix=name,
+            prefix_sep=separator,
+            drop_first=drop_first,
+            # Adds a null column at the end, depending on whether or not there are any.
+            dummy_na=has_nulls,
+            dtype=int,
```

Comment:
i'm a little torn on this one:
- setting 'uint8' so it matches the polars
- setting a signed integer type, because unsigned integers can be really unexpected to pandas users (pandas doesn't show types)

🤔 dunno...maybe `'int8'` is a good compromise?